gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/util/ObservationUtil.java

```diff
@@ -0,0 +1,216 @@
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.hl7.fhir.dstu3.model.BooleanType;
+import org.hl7.fhir.dstu3.model.CodeableConcept;
+import org.hl7.fhir.dstu3.model.Coding;
+import org.hl7.fhir.dstu3.model.Extension;
+import org.hl7.fhir.dstu3.model.InstantType;
+import org.hl7.fhir.dstu3.model.Observation;
+import org.hl7.fhir.dstu3.model.Period;
+import org.hl7.fhir.dstu3.model.Quantity;
+import org.hl7.fhir.dstu3.model.SimpleQuantity;
+import org.hl7.v3.CV;
+import org.hl7.v3.IVLPQ;
+import org.hl7.v3.IVLTS;
+import org.hl7.v3.PQ;
+import org.hl7.v3.RCMRMT030101UK04Author;
+import org.hl7.v3.RCMRMT030101UK04EhrComposition;
+import org.hl7.v3.RCMRMT030101UK04EhrExtract;
+import org.hl7.v3.RCMRMT030101UK04InterpretationRange;
+import org.hl7.v3.RCMRMT030101UK04ReferenceRange;
+import org.hl7.v3.TS;
+
+import uk.nhs.adaptors.pss.translator.mapper.QuantityMapper;
+
+public class ObservationUtil {
+    private static final String VALUE_QUANTITY_EXTENSION = "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-CareConnect"
+        + "-ValueApproximation-1";
+    private static final String CODING_SYSTEM = "http://hl7.org/fhir/v2/0078";
+
+    private static final QuantityMapper QUANTITY_MAPPER = new QuantityMapper();
+
+    public static Quantity getValueQuantity(Object value, CV uncertaintyCode) {
+        if (isValidValueQuantity(value)) {
+            Quantity valueQuantity;
+            if (value instanceof PQ) {
+                valueQuantity = QUANTITY_MAPPER.mapQuantity((PQ) value);
+            } else {
+                valueQuantity = QUANTITY_MAPPER.mapQuantity((IVLPQ) value);
+            }
+
+            if (uncertaintyCode != null) {
+                valueQuantity.getExtension().add(new Extension()
+                    .setUrl(VALUE_QUANTITY_EXTENSION)
+                    .setValue(new BooleanType().setValue(true)));
+            }
+
+            return valueQuantity;
+        }
+
+        return null;
+    }
+
+    public static CodeableConcept getInterpretation(CV interpretationCode) {
+        if (interpretationCode != null) {
+            var interpretationCodeableConcept = new CodeableConcept();
+
+            var code = interpretationCode.getCode();
+
+            interpretationCodeableConcept.getCoding().add(new Coding()
+                .setCode(getInterpretationCodeAbbreviation(code))
+                .setDisplay(getInterpretationDisplay(code))
+                .setSystem(CODING_SYSTEM));
+
+            if (StringUtils.isNotEmpty(interpretationCode.getOriginalText())) {
+                interpretationCodeableConcept.setText(interpretationCode.getOriginalText());
+            } else if (StringUtils.isNotEmpty(interpretationCode.getDisplayName())) {
+                interpretationCodeableConcept.setText(interpretationCode.getDisplayName());
+            }
+
+            return interpretationCodeableConcept;
+        }
+
+        return null;
```

Comment:
I think you could use CodeableConceptUtils, and the `getInterpretation` method could look like this:
```suggestion
       if (interpretationCode != null) {
            var code = interpretationCode.getCode();
            String text = null;
            if (StringUtils.isNotEmpty(interpretationCode.getOriginalText())) {
                text = interpretationCode.getOriginalText();
            } else if (StringUtils.isNotEmpty(interpretationCode.getDisplayName())) {
                text = interpretationCode.getDisplayName();
            }

            return CodeableConceptUtils.createCodeableConcept(
                getInterpretationCodeAbbreviation(code),
                CODING_SYSTEM,
                getInterpretationDisplay(code),
                text
            );
        }

        return null;
```